src/project_manager/ros_generic_run_step.cpp

```diff
@@ -124,6 +125,7 @@ void ROSGenericRunStep::run()
 
   //send roslaunch command
   terminal.sendText(command);
+#endif
```

Comment:
The "Application Output" pane (Alt+3) can actually not be used to provide input via `stdin` to an application, it only provides the output via `stdout`. So I think the only way to have I/O in Qt Creator would be via the external terminal.

However, what is the use case for stdin in `rosrun` and `roslaunch`? Do we really need a proper terminal for running nodes and launch files? If we go with the proposed solution (using the integrated run step), then terminal input would only be supported when a user chooses the external terminal.

At the moment, this PR would effectively inhibit any run steps in `ROSGenericRunStep::run` by just not running them. I currently see the following options:
1. disable all uses of `ROSGenericRunStep` until `qtermwidget` is ported to Qt6
2. manually launch an external terminal in `ROSGenericRunStep`
3. use the standard Qt Creator run steps with the above limitations

Since I am not using `rosrun` or `roslaunch` from Qt Creator, and I expect to switch to ROS2 within a year, I personally would prefer the first solution of just disabeling this feature for now. This would also be the one with the least work required.

@Levi-Armstrong What do you suggest how to go on with this?

Reply:
I think it may be better to leverage the external terminal until qtermwidget is available. I personally never using Qt Creator for running roslaunch or rosrun, but did use the terminal. I am fine with getting rid of the roslaunch and rosrun because this can be accomplished using the integrated runstep.

Reply:
The run steps, that are effectively deactivated by this PR, did not work in earlier versions before. They somehow broke because of API changes when I started porting the plugin to newer versions. So restoring this functionality will take more work than switching from qtermwidget5 to an external terminal. But this is something I would tackle in a dedicated PR.

If you are fine with the other changes, can we go on with merging this?

Reply:
If everything can be achieved using native Qt Creator capability the I would push that direction since it reduce the maintenance of this repo. I am fine with removing the roslaunch, rosrun and qtermwidget capability because I feel they most likely were never used.

Reply:
I also never used the rosrun etc functionality from the UI. Given that nobody complained that this functionality was missing since I ported the plugin to newer Qt Creator versions, I gather that it actually wasn't used by others either.

If I have the time, I will look into enabling this for ROS2. But the next time the API breaks more than what can be fixed with a search&replace, I am going to remove that functionality entirely from the codebase.